### ISSUE TITLE
Deprecate future encoder

### ISSUE THREAD

**Issue by jmcardon** [http4s/http4s#1757]
```scala
import cats.effect.IO

import scala.concurrent.ExecutionContext.Implicits._
import org.http4s._
import org.http4s.dsl.io._
import org.http4s.implicits._
import cats.implicits._

import scala.concurrent.Future
val service: HttpService[IO] = HttpService[IO]{
  case GET -> Root =>
    BadRequest(Future({
      println("hi1")
      "zz"
    }))
}

val service2: HttpService[IO] = HttpService[IO]{
  case GET -> Root =>
    Ok(Future({
      println("hi2")
      "zz"
    }))
}
//repl doesn't pick up our syntax :(
(service <+> service2).run(Request[IO](GET)).getOrElse(Response.notFound).unsafeRunSync
```

This prints both hi1 and h2. This is potentially terrible for business logic on top of the fact it clearly violates RT.

We need to deprecate `EntityEncoder.futureEncoder[F]`

**Comment by rossabaker** [comment]
Yeah, do it.  It exists for convenience, but I don't think the danger was properly appreciated.

**Comment by rossabaker** [comment]
Closed by #1759.
